app/models/embed/purl.rb

```diff
@@ -137,6 +138,30 @@ def initialize(msg = 'The requested PURL resource was not embeddable.')
         super
       end
     end
+
+    class Duration < ISO8601::Duration
+      def to_s
+        raise 'Duration#to_s does not support specifying durations in weeks' if atoms[:weeks] != 0
+        raise 'Duration#to_s does not support specifying negative durations' if atoms.values.detect { |val| val < 0 }
```

Comment:
i guess another option would be some sort of text indicator of the error, like returning `--`, though that seems like it might just confuse users.